pkg/storage/blockdev.go

```diff
@@ -264,52 +247,59 @@ const (
 	FAT32IDSize    = 4
 )
 
-func tryVFAT(devname string) (uuid string) {
-	log.Printf("try vfat")
+func tryVFAT(file *os.File) (string, error) {
 	var off int64
 
-	file, err := os.Open(devname)
-	if err != nil {
-		fmt.Println(err)
-		return ""
-	}
-	defer file.Close()
-
-	fileinfo, err := file.Stat()
-	if err != nil {
-		fmt.Println(err)
-		return ""
-	}
-	fmt.Printf("%s %d\n", fileinfo.Name(), fileinfo.Size())
-
 	// magic number
 	b := make([]byte, FAT32MagicSize)
 	off = 0 + FAT32MagicOff
-	_, err = file.ReadAt(b, off)
+	_, err := file.ReadAt(b, off)
 	if err != nil {
-		fmt.Println(err)
-		return ""
+		return "", err
 	}
 	magic := string(b)
-	fmt.Printf("magic: %s\n", magic)
 	if magic != FAT32Magic {
-		log.Printf("no vfat")
-		return ""
+		return "", fmt.Errorf("no vfat")
 	}
 
 	// filesystem UUID
 	b = make([]byte, FAT32IDSize)
 	off = 0 + FAT32IDOff
 	_, err = file.ReadAt(b, off)
 	if err != nil {
-		fmt.Println(err)
-		return ""
+		return "", err
+	}
+
+	return fmt.Sprintf("%02x%02x-%02x%02x", b[3], b[2], b[1], b[0]), nil
+}
+
+const (
+	XFSMagic     = "XFSB"
+	XFSMagicSize = 4
+	XFSUUIDOff   = 32
+	XFSUUIDSize  = 16
+)
+
+func tryXFS(file *os.File) (string, error) {
```

Comment:
I think maybe this should be `func tryXFS(file io.ReaderAt)` to make sure that nobody moves the file pointer around

Reply:
All the reads are now using readat.